compiler/rustc_trait_selection/src/solve/search_graph/mod.rs

```diff
@@ -210,6 +209,7 @@ impl<'tcx> SearchGraph<'tcx> {
                     &self.stack,
                     entry.goal,
                     entry.response,
+                    dep_node,
                 );
```

Comment:
here we move all participants in this cycle into the global cache but we only want to move the current head (i.e. the goal we just popped from the stack), all other goals should be discarded